adal/src/main/java/com/microsoft/aad/adal/Discovery.java

```diff
@@ -321,7 +327,22 @@ private URL buildQueryString(final String instance, final String authorizationEn
         return new URL(builder.build().toString());
     }
 
+    /**
+     * @return {@link ReentrantLock} for locking the network request queue.
+     */
+    private static ReentrantLock getLock() {
+        if (sLock == null) {
+            synchronized (Discovery.class) {
+                if (sLock == null) {
+                    sLock = new ReentrantLock();
+                }
+            }
+        }
+
+        return sLock;
+    }
+
     Set<String> getValidHosts() {
-        return VALID_HOSTS;
+        return AAD_WHITELIST_HOSTS;
```

Comment:
nit: should this be AAD_WHITELISTED_HOSTS? just curious

Reply:
updated